kernel/include/nano_internal.h

```diff
@@ -82,6 +82,56 @@ static inline unsigned int _Swap(unsigned int key)
 
 #ifdef CONFIG_USERSPACE
 /**
+ * @brief get the maximum number of free regions for memory domain partitions
+ *
+ * @return Max number of free regions
+ */
+extern u8_t _arch_get_max_partitions(void);
+
+/**
+ * @brief Configure MPU memory domain
+ *
+ * This function configures per thread memory domain reprogramming the MPU.
+ * Used when adding the currently running thread to a domain.
+ *
+ * @param thread thread info data structure.
+ */
+extern void _arch_configure_mem_domain(struct k_thread *thread);
+
+/**
+ * @brief Configure MPU memory partition
+ *
+ * This function configure the specific memory partition of a thread. Used
+ * when adding partitions to an active memory domain.
+ *
+ * @param thread thread info data structure.
+ * @param part_index partition index that needs to be configured.
+ */
+extern void _arch_configure_mem_partition(struct k_thread *thread,
+					  u32_t part_index);
+
+/**
+ * @brief Disable MPU memory partition
+ *
+ * This function disables the specific memory partition of a thread. Used
+ * when partitions are removed from an active memory domain.
+ *
+ * @param thread thread info data structure.
+ * @param part_index partition index that needs to be disabled.
+ */
+extern void _arch_disable_mem_partition(struct k_thread *thread,
+					u32_t part_index);
```

Comment:
Basically the same complaint: Is this the inverse opperation to _arch_configure_mem_partition()?  If so should be named as an opposite (i.e. add/remove, enable/disable, set/unset, etc...).

Also needs an error code return value if that part_index can be wrong at runtime (which it surely can).